src/tiled/commanddatamodel.cpp

```diff
@@ -433,6 +434,19 @@ bool CommandDataModel::dropMimeData(const QMimeData *data, Qt::DropAction, int,
     return false;
 }
 
+QKeySequence CommandDataModel::shortcut(const QModelIndex &index) const
+{
+    if (index.isValid())
+        return mCommands[index.row()].shortcut;
```

Comment:
Since you correctly pointed out this may crash, please keep the range check you had before.